torch/_inductor/codegen/common.py

```diff
@@ -36,6 +44,114 @@ def index_prevent_reordering(index: typing.List[sympy.Expr], index_vars, sizes):
     return [*index, sympy_dot(index_vars, FlexibleLayout.contiguous_strides(sizes))]
 
 
+def _data_type_propagation(sub_graph: torch.fx.Graph):
+    def propagate_node(node: torch.fx.Node):
+        _node: torch.fx.Node = node
+        if _node.target == "ops":
+            return False
```

Comment:
Check target in `ops_without_dtype` instead? Then, you don't have to explicitly check `ops_without_dtype` later?

Reply:
Yes, we can check `target in ops_without_dtype` here, but we still need to explicitly check ops_without_dtype later while propagate dtype from input nodes. 
```
        input_nodes = [
            n
            for n in inputs
            if isinstance(n, torch.fx.node.Node) and n.target not in ops_without_dtype
        ]
```
While `node.inputs.target = ['ops' ,'get_index', 'add'],` we need remove `ops` and `get_index` and only use `add` to decide the `node.dtype`